sys/net/routing/rpl/rpl_control_messages.c

```diff
@@ -840,8 +840,17 @@ void rpl_recv_DAO(void)
                       ipv6_addr_to_str(addr_str, IPV6_MAX_ADDR_STR_LEN, &rpl_opt_target_buf->target),
                       ipv6_addr_to_str(addr_str, IPV6_MAX_ADDR_STR_LEN, &ipv6_buf->srcaddr),
                       (rpl_opt_transit_buf->path_lifetime * my_dodag->lifetime_unit));
-                rpl_add_routing_entry(&rpl_opt_target_buf->target, &ipv6_buf->srcaddr,
-                                      rpl_opt_transit_buf->path_lifetime * my_dodag->lifetime_unit);
+
+                /* If we are the target we don't need to add a new FIB entry
+                 * This is a bit hack-ish, but sufficient for now
+                */
+                if ( my_address.uint32[3] != rpl_opt_target_buf->target.uint32[3] ) {
```

Comment:
@cgundogan the reason for this hack-ish check is that `my_addres` is the global IPv6 address, e.g. `abcd::ff:fe00:1234` and the `rpl_opt_target_buf->target` the link local address, e.g. `fe80::ff:fe00:1234`.
I'm also in favour using the proposed function, but I didn't find a way to ask for the link local address of the node here ( I also didn't searched for long ;) )